documentation/add/04_mpi_framework.md

```diff
@@ -1,43 +1,244 @@
+| `labIndex` | P | The number of parallel process, used to identify the particular parallel worker. |
+| `NumLabs` | P | Total number of workers in the parallel pool. |
+| **Common service methods:** | :: | *Methods, used by all frameworks regardless of the framework type:* |
+|`build_worker_init` | M | Generate ASCII string, used for initialization of all workers. See **Note^1.**
+| `add_persistent` | M | Helper method used to add persistent messages to the list of the messages, received from other labs. See **Note^2**
+| `check_get_persistent` | M | check if a message is a persistent message. See **Note^2**
+| `check_set_persistent` | M | check if the input message is a persistent message. See **Note^2**
+| **Abstract methods:** | :: | *The methods, exposing interface to MPI communications. Implementation is specific for each framework:* | 
+| `init_framework` | M | Given input data, necessary for framework initialization, make framework operational. E.g. for file-based framework it may be name of the folder and the number of the current parallel worker, when for MPI framework, this command wrapps for C++ *MPI_init* command. 
+|`send_message`| M | send message to a specified task. Unblocking or kind-of unblocking. 
+|`receive_message`| M | receive message from specified task. Depending on the message type (see **Note^3**) it can be blocking or unblocking message. | 
+|`probe_all`| M | list all messages existing in the system from the tasks requested. Non-blocking
+|`receive_all`| M | receive all messages directed to current node and originated from the tasks with id-s specified as input. Blocking if 
+
+|`finalize_all`||
 |`mess_name`||
-|`send_message`||
-|`receive_message`||
-|`probe_all`||
-|`send_all`||
-|`receive_all`||
-|`finalize_all`||
\ No newline at end of file
+
+#### Note^1 Common initialization
+
+Different frameworks launch different types of parallel processes but each process should know what task it need to perform. The job description can be very different in size, so it is better to provide such description by writing appropriate description files. Command line arguments are used to provide each parallel worker with information about the location of a folder, where job initialization information is placed on shared file system. As different frameworks and different operating system use different type of encoding and different protocols for transmitting command line arguments, the location information encoded into ASCII string using standard Java base64 encoder. The *build_worker_init* method performs the encoding of the input initialization information, namely initialization folder location and, for file-based transfer, the worker number, for further transfer of this information to parallel workers using command line arguments of correspondent job. 
```

Comment:
I have clarified this string as:

Reply:
To clarify, the only content of this message is the base64 encoded location of a folder containing the configuration file?

Reply:
Its the folder, the name of the framework used for interworkers communications and labNum for filebased communicator.  The name of the framework  actually can be moved out of this message and transmitted as the part of the starting message, which violates a bit current initialization logic (init: cluster-> JobExecutor->particular job), but this can be rethinked